tests/gce_virtual_machine_test.py

```diff
@@ -17,11 +17,20 @@
 import unittest
 import mock
 
+from perfkitbenchmarker import pkb  # noqa. Imported to create needed flags.
+from perfkitbenchmarker import benchmark_spec
+from perfkitbenchmarker import context
 from perfkitbenchmarker import vm_util
 from perfkitbenchmarker.providers.gcp import gce_virtual_machine
 
 
 class GCEPreemptibleVMFlagTestCase(unittest.TestCase):
+
+  def setUp(self):
+    # VM Creation depends on there being a BenchmarkSpec.
+    self.spec = benchmark_spec.BenchmarkSpec({}, 'name', 'benchmark_uid')
+    self.addCleanup(context.SetThreadBenchmarkSpec, None)
```

Comment:
This pattern looks asymmetric. Might be out of scope for this PR, but I suggest removing the auto-register from the spec constructor or adding a check in the constructor to make sure there isn't already a registered spec for the thread when the new one is constructed.